platform/core/src/studies/services/wado/retrieveMetadataLoaderAsync.js

```diff
@@ -0,0 +1,159 @@
+import { api } from 'dicomweb-client';
+import DICOMWeb from '../../../DICOMWeb/';
+import RetrieveMetadataLoader from './retrieveMetadataLoader';
+import { sortStudySeries, sortingCriteria } from '../../sortStudy';
+import getSeriesInfo from '../../getSeriesInfo';
+import * as StudyUtils from '../../studyUtils';
+
+/**
+ * Search series of a given study
+ * @param {DICOMwebClient} client Dicomweb client api
+ * @param {string} studyInstanceUID The Study Instance UID to search series from;
+ * @returns {Arrays} A list of Series Instances
+ */
+async function searchStudySeries(client, studyInstanceUID) {
+  const seriesList = await client.searchForSeries({ studyInstanceUID });
+  return seriesList;
+}
+
+/**
+ * Filter seriesList
+ * @param {Arrays} seriesList list of Series Instance UIDs
+ * @param {Object} filters object containing filter to be applied.
+ * @returns {Arrays} A list of Series Instances
+ */
+function filterStudySeries(seriesList = [], filters) {
+  let result = [...seriesList];
+  const { seriesInstanceUID: toCompare } = filters;
```

Comment:
So at this point, we at least have _some_ metadata for _every_ series in our study. I know in our `studyList` search, we're able to apply filters to fields. Can we apply a filter to the seriesInstanceUid field when we query?

See: https://www.dicomstandard.org/dicomweb/query-qido-rs/
First `key`/`value` matching

Once we have a list of all the `seriesInstanceUids` that qualify for our study + filter, we can then do the heavier metadata and instance calls? Also lets us skip filtering on the client (less data transmission)